### PR TITLE
Update history.py

### PR DESCRIPTION
**Pull request by sniffo**
> Added the self.time_stamps() method code for HistoryHdf5 class into the __init__() method so that time_stamps are only calculated once as self._time_stamps
> Edited self.time_stamps() method to return self._time_stamps
> Greatly improves efficiency of retrieve() method as time_stamps is not computed each time a new time-point is loaded.

### CODE REVIEW

**Comment by glyg** [comment]
Thanks a lot @sniffo I'll review that in december (when I have a bit more time)

**Comment by glyg** [comment]
Hi @sniffo I am working on the 1.0 release (see the README on the main branch)

**Comment by glyg** [comment]
If you have a moment, can you summarize the discussion on #273 in a jupyter notebook? I'll turn it into a test once I start working on coverage
G.

**Comment by sniffo** [comment]
Sure thing, what would you like in the notebook? Just highlight the differences in the two HistoryHdf5 implementations? 
- Tasnif

**Comment by glyg** [comment]
Hi @sniffo , Happy new year!


> Sure thing, what would you like in the notebook? Just highlight the differences in the two HistoryHdf5 implementations?

Yes, something as simple as possible illustrating the change.
